docs/qemu.md

```diff
@@ -53,6 +53,38 @@ gateway to the host network [SLiRP].
 Please note that ICMP won't work in the guest unless additional configuration is
 made, so the `ping` utility won't work.
 
+# 6. Remote debugging of Normal World applications
+If you need to debug a client application, using GDB in a remote debugging
+configuration may be useful. Remote debugging means `gdb` runs on your PC, where
+it can access the source code, while the program being debugged runs on the remote
+system (in this case, in the QEMU environment in normal world).
+Here is how to do that. On your PC, build with `GDBSERVER=y`:
+```
+$ cd build
+$ make -j8 run GDBSERVER=y
+[...]
+(qemu) c
+```
+Inside QEMU, run your application with gdbserver (for example `xtest 4002`):
+```
+# gdbserver :12345 xtest 4002
+Process xtest created; pid = 654
+Listening on port 12345
+
+```
+Back on your PC, in another terminal, start GDB and connect to the target:
+```
+$ ../out-br/host/bin/arm-buildroot-linux-gnueabihf-gdb
+(gdb) set sysroot ../out-br/host/arm-buildroot-linux-gnueabihf/sysroot
+(gdb) target remote :12345
```

Comment:
btw, maybe a bit off-topic (just my curiosity), but have you tried to debug using both gdbservers mutually (the one integrated into QEMU, and another running as a user app inside QEMU), for example, if you need to debug both CA and TA.

In case if the breakpoint is hit in the external gdbserver (integrated into QEMU), does another gdb instance loose connection to the internal one (by timeout)

Reply:
Not tried but each server has its own TCP port so they should be pretty much independent.